src/daft-dsl/src/functions/scalar.rs

```diff
@@ -132,28 +121,7 @@ pub trait ScalarUDF: Send + Sync + std::fmt::Debug + std::any::Any {
     ///     }
     /// }
     /// ```
-    #[allow(deprecated)]
-    fn function_args_to_field(
-        &self,
-        inputs: FunctionArgs<ExprRef>,
-        schema: &Schema,
-    ) -> DaftResult<Field> {
-        // for backwards compatibility we add a default implementation.
-        // TODO: move all existing implementations of `to_field` over to `function_args_to_field`
-        // Once that is done, we can name it back to `to_field`.
-        self.to_field(inputs.into_inner().as_slice(), schema)
-    }
-
-    #[deprecated = "use `function_args_to_field` instead"]
-    fn to_field(&self, inputs: &[ExprRef], schema: &Schema) -> DaftResult<Field> {
-        let inputs = inputs
-            .iter()
-            .map(|e| FunctionArg::unnamed(e.clone()))
-            .collect::<Vec<_>>();
-
-        let inputs = FunctionArgs::new_unchecked(inputs);
-        self.function_args_to_field(inputs, schema)
-    }
+    fn get_return_type(&self, args: FunctionArgs<ExprRef>, schema: &Schema) -> DaftResult<Field>;
```

Comment:
Technically it's returning a field, which in addition to a type, also has a name. Honestly it would make more sense to return a type, but I would not tackle this at the moment since it would require touching some of the iceberg partitioning logic which does change the expression name. For now, could we name it something like `get_return_field` instead?